tools/caas-tag-selector/tag-selector.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable no-underscore-dangle, import/no-unresolved */
+import 'https://da.live/nx/public/sl/components.js';
+import { LitElement, html } from 'https://da.live/nx/deps/lit/lit-core.min.js';
+import getStyle from 'https://da.live/nx/utils/styles.js';
+import DA_SDK from 'https://da.live/nx/utils/sdk.js';
+import { getTags, getAemRepo, getRootTags } from '../tags/tag-utils.js';
+
+const style = await getStyle(import.meta.url);
+const { context, token } = await DA_SDK.catch(() => null);
+const options = { headers: { Authorization: `Bearer ${token}` } };
+const collectionName = 'dx-tags/dx-caas';
+
+async function processRootTags(opts) {
+  const aemConfig = await getAemRepo(context, opts).catch(() => null);
+  if (!aemConfig || !aemConfig.aemRepo) {
+    console.log('Error in getting AemRepo');
+  }
+
+  const namespaces = aemConfig?.namespaces.split(',').map((namespace) => namespace.trim()) || [];
+  const rootTags = await getRootTags(namespaces, aemConfig, opts);
+
+  if (!rootTags || rootTags.length === 0) {
+    console.log('Error in getting RootTags');
+  }
+  return rootTags;
+}
+
+async function getTagCollection(root, name, opts) {
+  let collection = [];
+  for (const tag of root) {
+    const currentCollection = await getTags(tag.path, opts);
+    const firstTag = currentCollection[0];
+    if (firstTag && firstTag?.activeTag === name) {
+      collection = currentCollection;
+      return collection;
+    }
+  }
+  return collection;
+}
+class PageGeneratorCaaSTagSelector extends LitElement {
+  static properties = {
+    propCollection: { Type: Array },
+    _caasPPN: { state: true },
+    _contentTypes: { state: true },
+  };
+
+  constructor() {
+    super();
+    this._contentTypes = [];
+    this._caasPPN = [];
+  }
+
+  async connectedCallback() {
+    super.connectedCallback();
+    this.shadowRoot.adoptedStyleSheets = [style];
```

Comment:
You can set styles directly with [lit](https://lit.dev/docs/components/styles/).
```suggestion
    static styles = style;
```